internal/providers/pke/pkeworkflow/create_cluster.go

```diff
@@ -22,119 +22,51 @@ import (
-			"us-west-2":      "ami-02904d39ae2ed2a7e", // US West (Oregon).
-		}[region], nil
-	case constraint115.Check(kubeVersion):
-		return map[string]string{
-			"ap-east-1":      "ami-0c906c5a886224f2c", // Asia Pacific (Hong Kong).
-			"ap-northeast-1": "ami-048cd136c9d3752d6", // Asia Pacific (Tokyo).
-			"ap-northeast-2": "ami-0c2517ca3fd4e157e", // Asia Pacific (Seoul).
-			"ap-southeast-1": "ami-0186d86b776b8dc8b", // Asia Pacific (Mumbai).
-			"ap-southeast-2": "ami-0af43abdd3d123f8a", // Asia Pacific (Singapore).
-			"ap-south-1":     "ami-02ba2b7be8eb99b34", // Asia Pacific (Sydney).
-			"ca-central-1":   "ami-0644ae9cb4750efd5", // Canada (Central).
-			"eu-central-1":   "ami-0ccf38993187a12ff", // EU (Frankfurt).
-			"eu-north-1":     "ami-01ba2867145e54525", // EU (Stockholm).
-			"eu-west-1":      "ami-0e91b1f208d945645", // EU (Ireland).
-			"eu-west-2":      "ami-0fc75cd793034847b", // EU (London).
-			"eu-west-3":      "ami-0a362008b150c6f60", // EU (Paris).
-			"me-south-1":     "ami-0813b61c2db464b11", // Middle East (Bahrain).
-			"sa-east-1":      "ami-01f61154ea7d72f00", // South America (Sao Paulo)
-			"us-east-1":      "ami-09882d44c73fabe37", // US East (N. Virginia).
-			"us-east-2":      "ami-03a405f0e64a8cfe0", // US East (Ohio).
-			"us-west-1":      "ami-062b015f7ca2803f2", // US West (N. California).
-			"us-west-2":      "ami-09e2678e579e5b06f", // US West (Oregon).
-		}[region], nil
-	default:
-		return map[string]string{
-			"ap-east-1":      "ami-0c9680acbf35f26de", // Asia Pacific (Hong Kong).
-			"ap-northeast-1": "ami-0f13e8123146595b9", // Asia Pacific (Tokyo).
-			"ap-northeast-2": "ami-021015b95e7bdfbbe", // Asia Pacific (Seoul).
-			"ap-southeast-1": "ami-0382298e181ef5686", // Asia Pacific (Mumbai).
-			"ap-southeast-2": "ami-068231c38bc1a60f3", // Asia Pacific (Singapore).
-			"ap-south-1":     "ami-016ec067d44808c4f", // Asia Pacific (Sydney).
-			"ca-central-1":   "ami-0e06edb0102874198", // Canada (Central).
-			"eu-central-1":   "ami-0ec8d2a455affc7e4", // EU (Frankfurt).
-			"eu-north-1":     "ami-067696d723418ef5e", // EU (Stockholm).
-			"eu-west-1":      "ami-0214421b4d7aaecdd", // EU (Ireland).
-			"eu-west-2":      "ami-08576e40ab2877d2a", // EU (London).
-			"eu-west-3":      "ami-0cb72921b642a83ec", // EU (Paris).
-			"me-south-1":     "ami-0f5484e06a055b46d", // Middle East (Bahrain).
-			"sa-east-1":      "ami-08d90516f7c661b6b", // South America (Sao Paulo).
-			"us-east-1":      "ami-07079058aa890ee37", // US East (N. Virginia).
-			"us-east-2":      "ami-0faf98ec1c0e28a7e", // US East (Ohio).
-			"us-west-1":      "ami-0bef95b814eae1fc7", // US West (N. California).
-			"us-west-2":      "ami-0ca6e0198325b7be7", // US West (Oregon).
-		}[region], nil
+	if cloudInfoClient != nil {
+		ami, err := cloudInfoClient.PKEImageName("amazon", "pke", "ubuntu", kubeVersion.String(), pkeVersion, region)
+		if err != nil {
+			// fail silently
```

Comment:
Is this a good idea? Should we at least log it? Maybe retry the request?

Reply:
In this case one ami will be chosen from the static map a few lines later.